tests/sentry/api/endpoints/test_organization_details.py

```diff
@@ -783,6 +786,48 @@ def test_org_mapping_already_taken(self):
         OrganizationMapping.objects.create(organization_id=999, slug="taken", region_name="us")
         self.get_error_response(self.organization.slug, slug="taken", status_code=409)
 
+    def test_configure_auth_provider(self):
+        old_config = {"domain": "foo.com"}
+        new_config = {"domain": "bar.com"}
+        provider = "google"
+        self.get_success_response(
+            self.organization.slug,
+            method="put",
+            providerName=provider,
+            providerConfig=old_config,
+        )
+        auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+        assert auth_provider.provider == provider
+        assert auth_provider.config == {"domains": ["foo.com"], "version": DATA_VERSION}
+
+        self.get_success_response(
+            self.organization.slug,
+            method="put",
+            providerName=provider,
+            providerConfig=new_config,
+        )
+        auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+        assert auth_provider.provider == provider
+        assert auth_provider.config == {"domains": ["bar.com"], "version": DATA_VERSION}
+
+    def test_invalid_auth_provider_configuration(self):
```

Comment:
@schew2381 would you be able to add a test case whereby the `providerName` is a valid choice, but `providerConfig` is an invalid config?

Reply:
Added the following test:

```
  self.get_error_response(
      self.organization.slug,
      method="put",
      providerName="google",
      providerConfig={"invalid_domain": "foo.com"},
      status_code=500,
  )
```